### PR TITLE
[2.10] Show Cloud Credential Expiration information for Harvester Creds, add actions to renew

### LINKED ISSUE #11332: Allow users to easily see expiring harvester cloud credentials and to renew them
Epic: https://github.com/rancher/dashboard/issues/11256 SURE-8516

This is a forwardport issue for #11270, automatically created via [GitHub Actions workflow](https://github.com/rancher/dashboard/actions/runs/9762547786) initiated by @richard-cox

Original issue body:

Requires expiring label on cloud cred

- Global notification somewhere warning of expired / soon to expire harv cloud creds
- Clearly see in cloud cred table expired / soon to expire (see cluster dashboard certs list for pattern?)
  - Column for sorting, only if harv cloud cred present?
- Sub row in harv clusters list to warn that cluster has a cloud cred that's expiring
- Cloud Cred bulk action to renew (recreate kubeconfig, update cloud cred)

Edit: Blocked on https://github.com/rancher/rancher/issues/47241

### PR DESCRIPTION
**Pull request by richard-cox**
### Summary
Fixes #11332

### Occurred changes and/or fixed issues
- Cherry-picked PR using commits from https://github.com/rancher/dashboard/pull/12038
- Also cherry-pick of https://github.com/rancher/dashboard/pull/12203
- Also fix for cloud cred annotation collection (backport https://github.com/rancher/dashboard/pull/12230)

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/config/labels-annotations.js [hunk 1]
```diff
@@ -172,3 +172,7 @@ export const SYSTEM_LABELS = [
   'node.kubernetes.io',
   'egress.rke2.io'
 ];
+
+export const CLOUD_CREDENTIALS = [
+  'cattle.io/expiration-date'
```
Comment (richard-cox) [inline shell/config/labels-annotations.js:177]: this is now `cattle.io/expiration-timestamp`

Comment (richard-cox) [reply shell/config/labels-annotations.js:177]: now `rancher.io/expiration-timestamp`...

Comment (richard-cox) [reply shell/config/labels-annotations.js]: for reference https://github.com/rancher/rancher/pull/47488/files#diff-10c41bfb5e92c85f5c529be14f327d105ed94394ed74f8b3c2e46b523a506af0R27

**Review by momesgin — APPROVED** [review]
This PR cherry picks three other PRs, one is previously reviewed by me, and the other two by Francesco, so based on that I approve this

**Pull request merged**